generator/src/main/java/com/scottlogic/deg/generator/decisiontree/treepartitioning/TreePartitioner.java

```diff
@@ -47,29 +47,24 @@ public TreePartitioner() {
         final Map<RootLevelConstraint, Set<Field>> mapping = fieldMapper.mapConstraintsToFields(decisionTree);
 
         // each set of fields iterated here are constrained by a single root-level constraint/decision
-        for (RootLevelConstraint constraint : mapping.keySet()) {
+        // find which existing partitions this constraint/decision affects (if any)
+        // then, add new partition for this new constraint
+        // if there are any intersecting partitions, merge them with the new one
+        mapping.keySet().forEach(constraint -> {
             Set<Field> fields = mapping.get(constraint);
-
-            // find which existing partitions this constraint/decision affects (if any)
             final Set<Integer> existingIntersectingPartitions = fields
                 .stream()
                 .map(partitions::getPartitionId)
                 .distinct()
                 .filter(Objects::nonNull)
                 .collect(Collectors.toSet());
-
-            // then, add new partition for this new constraint
             final Integer partitionId = partitions.addPartition(fields, new HashSet<>(Collections.singletonList(constraint)));
-
-            // if there are any intersecting partitions, merge them with the new one
-            if (existingIntersectingPartitions.size() > 0) {
-                final Set<Integer> partitionsToMerge = new HashSet<>();
-                partitionsToMerge.add(partitionId);
-                partitionsToMerge.addAll(existingIntersectingPartitions);
-
-                partitions.mergePartitions(partitionsToMerge);
-            }
-        }
+            if (existingIntersectingPartitions.isEmpty()) return;
```

Comment:
opinion based, 
but on this project we try to avoid any if's without  a {} other than in autogenerated equals methods

Reply:
i personally find this way hard to read unless there is whitespace between it and the rest of the code